craft_parts/plugins/base.py

```diff
@@ -36,6 +37,17 @@
     from craft_parts import infos
 
 
+@dataclass(frozen=True)
```

Comment:
Given that we'll be creating a lot of these, it might be worthwhile to have:

```suggestion
@dataclass(frozen=True, slots=True)
```

Reply:
https://github.com/canonical/craft-parts/pull/931/commits/517e0ede0221c490f6df90106bdff9f6eff9c7bb